libraries/lib-stretching-sequence/ClipSegment.cpp

```diff
@@ -65,6 +65,12 @@ ClipSegment::ClipSegment(
 {
 }
 
+ClipSegment::~ClipSegment()
+{
+   mOnSemitoneShiftChangeSubscription.Reset();
```

Comment:
Why?

Reply:
To get the publisher to know a wee bit earlier that the subscription isn't valid anymore, hopefully before this object and its members gets destroyed ... but that reveals my lack of confidence that risking
```
[this](int cents) {
            mCentShift = cents;
            mUpdateCentShift = true;
         }
```
to be called after `this` is destroyed` (and hence `mCentShift` (`bool`) and `mUpdateCentShift` (`atomic<bool>`)) is safe. I couldn't crash Audacity anymore when stress-testing this on Windows and Mac, but I don't know the theoretical justification, if it does exist. Input welcome.

Reply:
That depends on the data layout, but ok